app/controllers/courses_controller.rb

```diff
@@ -58,7 +58,7 @@ def statistics
       if current_user.id == @teacher_id
         @teacher = true
       elsif current_user.student?
-        @student = @terms.first.students.find_by user_id: current_user.id
+        @student = @terms.collect { |term| term.students.where(user_id: current_user.id) }.any?
```

Comment:
Well, maybe still not. It doesn't seem to be possible to access `course` or `course_id` from `Student` model. I would surely use `Student.where(params)` if we only needed to find student by id, but I suppose that their presence in the course should also be checked.